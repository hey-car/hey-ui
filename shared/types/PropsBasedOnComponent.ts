import React from 'react';

/**
 * `PropsBasedOnComponent` - This interface inherits props from a designated Component
 */
export interface PropsBasedOnComponent<
  ComponentBaseProps,
  DefaultElement extends React.ElementType,
> {
  <Component extends React.ElementType = DefaultElement>(
    props: {
      /**
       * The component used for the root node.
       * Either a string to use a HTML element or a component.
       */
      component?: Component;
    } & ComponentBaseProps &
      Omit<React.ComponentPropsWithoutRef<Component>, keyof ComponentBaseProps>,
  ): JSX.Element | null;
}
