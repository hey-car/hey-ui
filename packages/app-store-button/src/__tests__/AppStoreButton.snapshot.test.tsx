import React from 'react';
import { render } from '@testing-library/react';

import AppStoreButton from '../AppStoreButton';
import { ComponentProps } from '../AppStoreButton.types';

const DEFAULT_DATA: ComponentProps = {
  href: 'https://apps.apple.com/uk/app/heycar/id1490142063',
  lang: 'en-EN',
  store: 'apple',
};

describe('AppStoreButton Snapshots tests', () => {
  describe('Apple', () => {
    it('should render default button in german', () => {
      expect(
        render(<AppStoreButton {...DEFAULT_DATA} lang="de-DE" />),
      ).toMatchSnapshot();
    });

    it('should render default button in english', () => {
      expect(render(<AppStoreButton {...DEFAULT_DATA} />)).toMatchSnapshot();
    });

    it('should render default button in spanish', () => {
      expect(
        render(<AppStoreButton {...DEFAULT_DATA} lang="es-ES" />),
      ).toMatchSnapshot();
    });

    it('should render default button in french', () => {
      expect(
        render(<AppStoreButton {...DEFAULT_DATA} lang="fr-FR" />),
      ).toMatchSnapshot();
    });

    it('should render alt button in german', () => {
      expect(
        render(<AppStoreButton {...DEFAULT_DATA} lang="de-DE" theme="alt" />),
      ).toMatchSnapshot();
    });

    it('should render alt button in english', () => {
      expect(
        render(<AppStoreButton {...DEFAULT_DATA} theme="alt" />),
      ).toMatchSnapshot();
    });

    it('should render alt button in spanish', () => {
      expect(
        render(<AppStoreButton {...DEFAULT_DATA} lang="es-ES" theme="alt" />),
      ).toMatchSnapshot();
    });

    it('should render alt button in french', () => {
      expect(
        render(<AppStoreButton {...DEFAULT_DATA} lang="fr-FR" theme="alt" />),
      ).toMatchSnapshot();
    });
  });

  describe('Google', () => {
    const GOOGLE_DATA: ComponentProps = {
      href: 'https://apps.apple.com/uk/app/heycar/id1490142063',
      lang: 'en-EN',
      store: 'google',
    };

    it('should render default button in german', () => {
      expect(
        render(<AppStoreButton {...GOOGLE_DATA} lang="de-DE" />),
      ).toMatchSnapshot();
    });

    it('should render default button in english', () => {
      expect(render(<AppStoreButton {...GOOGLE_DATA} />)).toMatchSnapshot();
    });

    it('should render default button in spanish', () => {
      expect(
        render(<AppStoreButton {...GOOGLE_DATA} lang="es-ES" />),
      ).toMatchSnapshot();
    });

    it('should render default button in french', () => {
      expect(
        render(<AppStoreButton {...GOOGLE_DATA} lang="fr-FR" />),
      ).toMatchSnapshot();
    });
  });
});
