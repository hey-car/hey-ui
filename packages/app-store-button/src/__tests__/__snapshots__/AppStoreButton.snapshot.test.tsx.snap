// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppStoreButton Snapshots tests Apple should render alt button in english 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        aria-label="Download on the App Store"
        class="wrapper apple "
        href="https://apps.apple.com/uk/app/heycar/id1490142063"
        rel="nofollow"
        target="_blank"
      >
        <img
          alt="Download on the App Store"
          src=""
        />
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      aria-label="Download on the App Store"
      class="wrapper apple "
      href="https://apps.apple.com/uk/app/heycar/id1490142063"
      rel="nofollow"
      target="_blank"
    >
      <img
        alt="Download on the App Store"
        src=""
      />
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AppStoreButton Snapshots tests Apple should render alt button in french 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        aria-label="Télécharger dans l'App Store"
        class="wrapper apple "
        href="https://apps.apple.com/uk/app/heycar/id1490142063"
        rel="nofollow"
        target="_blank"
      >
        <img
          alt="Télécharger dans l'App Store"
          src=""
        />
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      aria-label="Télécharger dans l'App Store"
      class="wrapper apple "
      href="https://apps.apple.com/uk/app/heycar/id1490142063"
      rel="nofollow"
      target="_blank"
    >
      <img
        alt="Télécharger dans l'App Store"
        src=""
      />
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AppStoreButton Snapshots tests Apple should render alt button in german 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        aria-label="Laden im App Store"
        class="wrapper apple "
        href="https://apps.apple.com/uk/app/heycar/id1490142063"
        rel="nofollow"
        target="_blank"
      >
        <img
          alt="Laden im App Store"
          src=""
        />
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      aria-label="Laden im App Store"
      class="wrapper apple "
      href="https://apps.apple.com/uk/app/heycar/id1490142063"
      rel="nofollow"
      target="_blank"
    >
      <img
        alt="Laden im App Store"
        src=""
      />
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AppStoreButton Snapshots tests Apple should render alt button in spanish 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        aria-label="Consíguelo en el App Store"
        class="wrapper apple "
        href="https://apps.apple.com/uk/app/heycar/id1490142063"
        rel="nofollow"
        target="_blank"
      >
        <img
          alt="Consíguelo en el App Store"
          src=""
        />
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      aria-label="Consíguelo en el App Store"
      class="wrapper apple "
      href="https://apps.apple.com/uk/app/heycar/id1490142063"
      rel="nofollow"
      target="_blank"
    >
      <img
        alt="Consíguelo en el App Store"
        src=""
      />
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AppStoreButton Snapshots tests Apple should render default button in english 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        aria-label="Download on the App Store"
        class="wrapper apple "
        href="https://apps.apple.com/uk/app/heycar/id1490142063"
        rel="nofollow"
        target="_blank"
      >
        <img
          alt="Download on the App Store"
          src=""
        />
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      aria-label="Download on the App Store"
      class="wrapper apple "
      href="https://apps.apple.com/uk/app/heycar/id1490142063"
      rel="nofollow"
      target="_blank"
    >
      <img
        alt="Download on the App Store"
        src=""
      />
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AppStoreButton Snapshots tests Apple should render default button in french 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        aria-label="Télécharger dans l'App Store"
        class="wrapper apple "
        href="https://apps.apple.com/uk/app/heycar/id1490142063"
        rel="nofollow"
        target="_blank"
      >
        <img
          alt="Télécharger dans l'App Store"
          src=""
        />
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      aria-label="Télécharger dans l'App Store"
      class="wrapper apple "
      href="https://apps.apple.com/uk/app/heycar/id1490142063"
      rel="nofollow"
      target="_blank"
    >
      <img
        alt="Télécharger dans l'App Store"
        src=""
      />
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AppStoreButton Snapshots tests Apple should render default button in german 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        aria-label="Laden im App Store"
        class="wrapper apple "
        href="https://apps.apple.com/uk/app/heycar/id1490142063"
        rel="nofollow"
        target="_blank"
      >
        <img
          alt="Laden im App Store"
          src=""
        />
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      aria-label="Laden im App Store"
      class="wrapper apple "
      href="https://apps.apple.com/uk/app/heycar/id1490142063"
      rel="nofollow"
      target="_blank"
    >
      <img
        alt="Laden im App Store"
        src=""
      />
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AppStoreButton Snapshots tests Apple should render default button in spanish 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        aria-label="Consíguelo en el App Store"
        class="wrapper apple "
        href="https://apps.apple.com/uk/app/heycar/id1490142063"
        rel="nofollow"
        target="_blank"
      >
        <img
          alt="Consíguelo en el App Store"
          src=""
        />
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      aria-label="Consíguelo en el App Store"
      class="wrapper apple "
      href="https://apps.apple.com/uk/app/heycar/id1490142063"
      rel="nofollow"
      target="_blank"
    >
      <img
        alt="Consíguelo en el App Store"
        src=""
      />
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AppStoreButton Snapshots tests Google should render default button in english 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        aria-label="Get it on Google Play"
        class="wrapper google "
        href="https://apps.apple.com/uk/app/heycar/id1490142063"
        rel="nofollow"
        target="_blank"
      >
        <img
          alt="Get it on Google Play"
          src=""
        />
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      aria-label="Get it on Google Play"
      class="wrapper google "
      href="https://apps.apple.com/uk/app/heycar/id1490142063"
      rel="nofollow"
      target="_blank"
    >
      <img
        alt="Get it on Google Play"
        src=""
      />
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AppStoreButton Snapshots tests Google should render default button in french 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        aria-label="Disponible sur Google Play"
        class="wrapper google "
        href="https://apps.apple.com/uk/app/heycar/id1490142063"
        rel="nofollow"
        target="_blank"
      >
        <img
          alt="Disponible sur Google Play"
          src=""
        />
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      aria-label="Disponible sur Google Play"
      class="wrapper google "
      href="https://apps.apple.com/uk/app/heycar/id1490142063"
      rel="nofollow"
      target="_blank"
    >
      <img
        alt="Disponible sur Google Play"
        src=""
      />
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AppStoreButton Snapshots tests Google should render default button in german 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        aria-label="Jetzt bei Google Play"
        class="wrapper google "
        href="https://apps.apple.com/uk/app/heycar/id1490142063"
        rel="nofollow"
        target="_blank"
      >
        <img
          alt="Jetzt bei Google Play"
          src=""
        />
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      aria-label="Jetzt bei Google Play"
      class="wrapper google "
      href="https://apps.apple.com/uk/app/heycar/id1490142063"
      rel="nofollow"
      target="_blank"
    >
      <img
        alt="Jetzt bei Google Play"
        src=""
      />
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AppStoreButton Snapshots tests Google should render default button in spanish 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        aria-label="Disponible en Google Play"
        class="wrapper google "
        href="https://apps.apple.com/uk/app/heycar/id1490142063"
        rel="nofollow"
        target="_blank"
      >
        <img
          alt="Disponible en Google Play"
          src=""
        />
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      aria-label="Disponible en Google Play"
      class="wrapper google "
      href="https://apps.apple.com/uk/app/heycar/id1490142063"
      rel="nofollow"
      target="_blank"
    >
      <img
        alt="Disponible en Google Play"
        src=""
      />
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
